src/item.cpp

```diff
@@ -4831,20 +4834,37 @@ void item::melee_combat_info( std::vector<iteminfo> &info, const iteminfo_query
     }
 
     if( dmg_bash || dmg_cut || dmg_stab ) {
-        if( parts->test( iteminfo_parts::BASE_TOHIT ) ) {
+        int stam;
+        float stam_pct;
+        std::map<std::string, double> dps_data;
+
+        bool base_tohit = parts->test( iteminfo_parts::BASE_TOHIT );
+        bool base_moves = parts->test( iteminfo_parts::BASE_MOVES );
+        bool base_dps = parts->test( iteminfo_parts::BASE_DPS );
+        bool base_stamina = parts->test( iteminfo_parts::BASE_STAMINA );
+        bool base_dpstam = parts->test( iteminfo_parts::BASE_DPSTAM );
+
+        if( base_stamina || base_dpstam ) {
```

Comment:
you have an error because stam and stam_pct aren't initialized above (to something like 0 and 0.0f)
so if you don't enter this conditional they have undefined values.

Reply:
huh, okay. thanks